src/Encoder.c

```diff
@@ -4635,13 +4635,13 @@ ZYDIS_EXPORT ZyanStatus ZydisEncoderDecodedInstructionToEncoderRequest(
     }
     request->allowed_encodings = 1 << instruction->encoding;
 
-    if ((operand_count > ZYDIS_ENCODER_MAX_OPERANDS) || 
-        (operand_count > instruction->operand_count_visible))
+    if ((operand_count_visible > ZYDIS_ENCODER_MAX_OPERANDS) ||
+        (operand_count_visible > instruction->operand_count_visible))
```

Comment:
@mappzor wondering whether this should really be `==`. Does it ever make sense to pass a lower number? It's an API breaking chance, so I didn't apply it, but might make sense to change this in v5?

Reply:
Those checks were changed here: https://github.com/zyantific/zydis/commit/875c8a75344d0ad71b02bcd2cc5e3acb87e7b003#diff-b27231e6044d9f536c52c09cb128aee56c82926c92676b480bd438df1f96787aL4407

It was surely the safest change to make considering how old code used to work but I don't think it ever makes sense to pass a lower number. Even if there was such case I'd consider it a bad way to use the API. It's assumed that result of decoding is passed without changes to `ZydisEncoderDecodedInstructionToEncoderRequest` and eventual adjustments to operand count, types, etc. are applied manually to resulting request. I'm in favor of `==` check in future version.

Reply:
I think my PR with the ZydisRegisterGetLargestEnclosing was already breaking and APX definitely is, so I'd say "go for it"!

We might have to release a new major version soon anyways.

Reply:
Yeah. The `ZydisRegisterGetLargestEnclosing` was indeed API breaking, though in a similar way as this one: you'd have to use the function in a really weird way to realize it. But that doesn't change that it was API breaking.

I adjusted the code to check with `==`. I further moved the check up and combined it with the other trivial argument validation.